backend/entityservice/tasks/comparing.py

```diff
@@ -220,7 +222,13 @@ def aggregate_comparisons(similarity_result_files, project_id, run_id, parent_sp
     files = []
     data_size = 0
 
-    for num, filename in similarity_result_files:
+    for res in similarity_result_files:
+        if res is None:
+            log.warning("Missing results during aggregation. Stopping processing.")
+            return
```

Comment:
You do a silent fail here, as well as a few lines up. This would result in the run status staying 'running' forever. I think that whenever we do not intend to actually compute a result, we should set the status to 'error'...